src/lib/sql/sql_identifier_resolver.hpp

```diff
@@ -14,7 +14,7 @@ class AbstractExpression;
 
 struct SQLIdentifierContextEntry final {
   std::shared_ptr<AbstractExpression> expression;
-  std::optional<SQLIdentifier> identifier;
+  std::vector<SQLIdentifier> identifiers;
```

Comment:
Can multiple identifiers of an expression have different `table_name`s? if not, wouldn't it be cleaner to make this 
```c++
struct SQLIdentifierContextEntry final {
  std::shared_ptr<AbstractExpression> expression;
  std::optional<std::string> table_name;
  std::vector<std::string> column_names;
}
```

if it is possible that there are multiple table names for one expression, please add SQLiteTestRunner and SQLTranslatorTest - queries for it.

Reply:
Having multiple identifiers for the same expression with different `table_name`s could work for a query like:
```sql
SELECT R.a, S.a FROM mixed AS R, mixed AS S
```
However, the `SQLTranslator` does not support this behavior since it creates two `StoredTableNode`s in `_translate_stored_table()`. Therefore, `R.a` and `S.a` are identifiers for different expressions. Do you think that changing the SQLTranslator makes sense?
I also added tests for the mentioned query.

Reply:
On a general note: As long as we don't break anything, we don't necessarily have to cover edge cases that we don't see in the TPC-DS

Reply:
> However, the SQLTranslator does not support this behavior since it creates two StoredTableNodes in _translate_stored_table().

That is fundamentally intended behavior. R.a and S.a are different columns/expressions in this case. They just happen to come from the same table. Think of `SELECT R.a, S.a FROM mixed AS R, mixed AS S WHERE S.a > 5` - it is obvious that R.a and S.a now contain different data - so they cannot possibly be the same expression/column

Reply:
Thanks for the clarification. Handling these columns identically would not help us implementing the TPC-DS. Thus, I won't change the `SQLTranslator`.

Reply:
why is this resolved? I believe the answer to my original question ("Can multiple identifiers of an expression have different table_names?") was "No", so wouldn't the layout of `SQLIdentifierContextEntry` that I proposed be cleaner?

Reply:
@mrzzzrm I checked the other open points, which are addressed. With this one, I am not sure enough to discard your change request. Could you have a look at https://github.com/hyrise/hyrise/pull/1693/commits/67fb153dd8cd8cffdd105f062c2988eb0dcba1df

Reply:
Yes, this commit addresses your suggestion @mrzzzrm. (Sorry for resolving the discussion too soon, though.) As a consequence, we can now run queries, such as `SELECT a AS x FROM int_float GROUP BY int_float.x`, which does not look right in my eyes but is unambiguous.

Reply:
Ok, I believe that's acceptable @mrks ?

Reply:
Doesn’t feel completely right, but I don’t expect it to hurt anywhere.